src/python/grpcio_tests/tests/unit/_metadata_flags_test.py

```diff
@@ -101,8 +102,9 @@ def service(self, handler_call_details):
 
 def create_dummy_channel():
     """Creating dummy channels is a workaround for retries"""
-    with bound_socket() as (host, port):
-        return grpc.insecure_channel('{}:{}'.format(host, port))
+    host, port, sock = get_socket()
+    sock.close()
```

Comment:
Without `SO_REUSEPORT`, there's a race condition here, right? The thread gets preempted between `sock.close()` and `grpc.insecure_channel`, and then the kernel allocates `port` to some other process on the system. Then `grpc.insecure_channel` fails to bind. I'm not sure what the failure rate should be though. Can you run with `--runs_per_test=10000` to find out whether or not this is worth worrying about?